### DISCUSSION TITLE
Korzyści ACR build

### DISCUSSION THREAD

**Discussion by sbugalski** [AzureCommunityPL/group#313] [Pomoc] [answered]
Cześć,

Rozważam, rozważam korzyści z wykorzystania budowania obrazów za pomocą ACR w porównaniu do _typowych_ pipelinów budujących uruchamianych za pomocą AzDO, GA, Jenkins.

Jedyna istotna różnica, którą widzę na pierwszy rzut oka to możliwość automatycznego przebudowania obrazów w sytuacji powstania nowej wersji obrazu bazowego.

Czy są jeszcze jakieś istotne zalety, których nie widzę?

**Comment by kamil-mrzyglod** [comment, marked as answer]
Ale to tutaj rozróżnijmy budowanie obrazów poprzez ACR za pomocą `az acr build` oraz automatyczne przebudowanie obrazu bazując na detekcji zmian zależności. Nie do końca czuję jak chcesz zastąpić swoje pipeline'y funkcjonalnością ACR.

**Reply by sbugalski** [reply]
Szukam odpowiedzi, czy ACR build > standardowy pipeline do budowania obrazów i wypchnięcie obrazów.

Automatyczne przebudowywanie obrazów na podstawie zmian to jedna z korzyści. Szukam kolejnych

**Reply by kamil-mrzyglod** [reply]
IMHO to jest jedyna korzyć. Konieczność utrzymywania kolejnego narzędzia, brak kontroli nad środowiskiem, nauka kolejnej składni nigdy mnie nie przekonały. Ja w większości wypadków i tak musiałem na podstawie takiego update'u zależności triggerować tradycyjne pipeline'y CI chociażby po to, aby przetestować zmianę i ją wdrożyć. Tak więc dla mnie przerost formy nad treścią.

**Reply by alipek** [reply]
Dla mnie plusem jest budowanie na wiele architektur. Nie muszę tworzyć środowiska CI żeby zbudować obraz np. dla arm64.

**Reply by sbugalski** [reply]
Dzięki Panowie to mi daje wystarczające informacje na podjęcie decyzji :)
